charts/tidb-operator/templates/admission-controller-deployment.yaml

```diff
@@ -0,0 +1,101 @@
+{{- if .Values.admissionController.create }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: admission-controller-certs
+data:
+  cert.pem: {{ .Values.admissionController.cert }}
+  key.pem: {{ .Values.admissionController.key }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: admission-controller-svc
+  labels:
+    app.kubernetes.io/name: {{ template "chart.name" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: admission-controller
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  ports:
+  - port: 443
+    targetPort: 443
+  selector:
+    app.kubernetes.io/name: {{ template "chart.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: admission-controller
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tidb-admission-controller-deployment
```

Comment:
I don't want to argue about naming style, bug add `-deployment` suffix to a deployment object is wired, so does the `tidb-admission-controller-pod` and `admission-controller-svc` and `admission-controller-cfg`